src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2175,6 +2201,23 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
 
             Debug.Assert(false, "Missing case in lambda conversion error reporting");
             diagnostics.Add(ErrorCode.ERR_InternalError, syntax.Location);
+
+            static void reportDefaultParameterMismatchError(BindingDiagnosticBag diagnostics, Location lambdaParameterLocation, int paramIdx, ConstantValue lambdaDefault, ConstantValue? delegateDefault)
```

Comment:
>reportDefaultParameterMismatchError [](http://example.com/codeflow?start=24&length=35)

Consider inlining this local function. It seems simple enough and there is only one caller.